<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro">


 <xacro:macro name="modelidar_ros2_control" params="name prefix use_mock_hardware">

<!-- <parameters>$(find articubot_one)/config/my_controllers.yaml</parameters> -->


<ros2_control name="${name}" type="system">
      <xacro:unless value="${use_mock_hardware}">
        <hardware>
          <plugin>modelidar/ModelidarSystemHardware</plugin>
          <param name="left_wheel_joint">virtual_left_wheel_joint</param> <!-- this value is also in controllers yaml file -->
          <param name="right_wheel_joint">virtual_right_wheel_joint</param> <!-- this value is also in controllers yaml file -->
          <param name="loop_rate">30</param>
          <param name="device">/dev/ttyACM0</param>
          <param name="baud_rate">115200</param>
          <param name="timeout">1000</param>
          <param name="enc_counts_per_rev">544</param> <!-- 16x34 for encoder DAGU RS003-Enco -->
        </hardware>
      </xacro:unless>
      <xacro:if value="${use_mock_hardware}">
        <hardware>
          <plugin>mock_components/GenericSystem</plugin>
          <param name="calculate_dynamics">true</param>
        </hardware>
      </xacro:if>
      <joint name="${prefix}virtual_left_wheel_joint">
        <command_interface name="velocity"/>
        <state_interface name="position"/>
        <state_interface name="velocity"/>
      </joint>
      <joint name="${prefix}virtual_right_wheel_joint">
        <command_interface name="velocity"/>
        <state_interface name="position"/>
        <state_interface name="velocity"/>
      </joint>
    </ros2_control>


  </xacro:macro>
</robot>